.circleci/config.yml

```diff
@@ -0,0 +1,45 @@
+version: 2
+jobs:
+  build:
+    docker:
+      - image: circleci/node:8.12.0
+    working_directory: ~/lansingcodes/lubot
+    environment:
+      TZ: America/Detroit
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v1-dep-{{ .Branch }}-
+            - v1-dep-master-
+            - v1-dep-
+      - run:
+          command: |
+            npm install
+            npm test
+      - save_cache:
+          key: v1-dep-{{ .Branch }}-{{ epoch }}
+          paths:
+            - ./node_modules
+  deploy:
+    machine: true
+    steps:
+      - checkout
+      - run:
+          command: |
+            pip install --upgrade awscli
+            $(aws ecr get-login --region us-east-1 --no-include-email)
+            docker build -t lubot .
+            docker tag lubot:latest $DOCKER_REPOSITORY:latest
+            if [[ "${CIRCLE_BRANCH}" == "master" ]]; then
+              docker push $DOCKER_REPOSITORY:latest
+              ssh lubot@aws-docker-1.atomaka.com "bash /home/lubot/deploy.sh"
+            fi
+workflows:
+  version: 2
+  test-and-deploy:
+    jobs:
+      - build
+      - deploy:
+          requires:
+            - build
```

Comment:
Does it make sense to add the lines below so `deploy` only triggers on changes to the `master` branch? I don't understand the `docker` and other commands to know if there's value in running them for pushes to all branches...

Reply:
I think there is value in running the docker build commands for every push. I had thought about this, and based on your comment I think I will go ahead and split the job so that it can make it more obvious that the deploy is a workflow step that is skipped for non-master branches

Reply:
Okay, I'll keep my eye out for an update to the PR. If you're chomping at the bit to move this along, poke me in Slack, too. 🙂

Reply:
With a little more understanding of Docker myself, I split the jobs so that for master it does the full build, tag, push, deploy, and for any other branch it just does a Docker build, basically to just kick the tires

Reply:
what I would prefer to do, is have the dockerbuild job produce an artifact that is used in the deploy job, but it doesn't seem straight forward. If someone know how to do that, it may be prefereable

Reply:
Yeah, reusing artifacts across jobs is lacking some features that would make your suggestion a lot easier. I like this and think it's a great compromise! 🤘